src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Automation/VsProject.cs

```diff
@@ -131,5 +134,35 @@ public string GetUniqueFilename(object pDispatch, string bstrRoot, string bstrDe
         {
             return _vsProject.GetUniqueFilename(pDispatch, bstrRoot, bstrDesiredExt);
         }
+
+        #region IConnectionPointContainer
+
+        public void EnumConnectionPoints(out IEnumConnectionPoints ppEnum)
+        {
+            ppEnum = null;
+            (_vsProject as IConnectionPointContainer)?.EnumConnectionPoints(out ppEnum);
```

Comment:
Under what circumstance will _vsProject not implement IConnectPointContainer?

Reply:
_vsProject doesn't support it before the CPS insertion that contains the CPS side of the work.
Basically, this allows dotnet project system to be used in combination with CPS versions that still don't have the new feature yet.

Reply:
But does it? We now say we support IConnectionPointContainer (by implemented the interface) - but don't actually work. If we insert this - we're going to be working in circumstances that we've never tested.

Reply:
I did some testing using that combination and it seemed to be working fine.
App Designer already handles the case where the connection point is Null:
https://github.com/dotnet/project-system/blob/master/src/Microsoft.VisualStudio.AppDesigner/interop/NativeMethods.vb#L121

In Debug builds it does trigger the assert that had the NullReference exception - which I also fixed.

Reply:
@adrianvmsft What am I missing? ConnectionPoint wouldn't be null in this case because "we" implement IConnectionPointContainer?

Reply:
Maybe ping me offline - I don't think I'm getting my point across, or I'm misunderstanding how this works.

Reply:
cpc.FindConnectionPoint returns null connection point here:
https://github.com/dotnet/project-system/blob/master/src/Microsoft.VisualStudio.AppDesigner/interop/NativeMethods.vb#L116

I did some additional validation, the assertion is new but doesn't seem to cause side effects.
Other options:
1. Wait until the CPS changes are in before checking in this change - arguably also changing the cast to not use as.
2. Implement custom query interface to not support the interface if the CPS object doesn't support it; considering this is a short term problem (until the CPS changes are in), I don't think it's worth proceeding this path

Reply:
Ah, that's what I was missing. Thanks!